core/app/models/spree/adjustment.rb

```diff
@@ -177,11 +174,6 @@ def update!(target = nil)
 
     private
 
-    def update_adjustable_adjustment_total
-      # Cause adjustable's total to be recalculated
-      ItemAdjustments.new(adjustable).update
-    end
-
     def require_promotion_code?
       promotion? && source.promotion.codes.any?
     end
```

Comment:
I'm thinking perhaps we should leave that for a separate PR?  If it was relying on the after destroy then it was leaving the order in an inconsistent state anyway, because it didn't update the order-level totals.